src/intersection/compute_intersection.cc

```diff
@@ -0,0 +1,1139 @@
+    // set pointers to Plucker coordinates for 1D-3D
+    // distribute Plucker products - CI13
+    for(unsigned int triangle_side = 0; triangle_side < RefElement<2>::n_sides; triangle_side++){
+        for(unsigned int line = 0; line < RefElement<3>::n_lines; line++){
+            CI13[triangle_side].set_plucker_product(
+                    plucker_products_[triangle_side * RefElement<3>::n_lines + line],
+                    line);
+            CI12[line].set_plucker_product(
+                    plucker_products_[triangle_side * RefElement<3>::n_lines + line],
+                    triangle_side);
+            
+            CI13[triangle_side].set_pc_tetrahedron(plucker_coordinates_tetrahedron[line],line);
+            CI12[line].set_pc_triangle(plucker_coordinates_triangle_[triangle_side],triangle_side);
+        }
+        CI13[triangle_side].set_pc_abscissa(plucker_coordinates_triangle_[triangle_side]);
+        CI13[triangle_side].init();
+    }
+    
+    // set pointers to Plucker coordinates for 1D-2D
+    for(unsigned int line = 0; line < RefElement<3>::n_lines; line++)
+        CI12[line].set_pc_abscissa(plucker_coordinates_tetrahedron[line]);
+
+};
+
+void ComputeIntersection<Simplex<2>, Simplex<3>>::compute(IntersectionAux< 2 , 3  >& intersection, 
+                                                          std::vector<unsigned int> &prolongation_table){
+
+	std::vector<IntersectionPointAux<1,2>> IP12s;
+	std::vector<IntersectionPointAux<1,3>> IP13s;
+	unsigned int pocet_13_pruniku;
+
+	for(unsigned int triangle_line = 0; triangle_line < RefElement<2>::n_lines; triangle_line++){    // go through triangle lines
+// 		DBGMSG("2d-3d triangle side %d\n",triangle_line);
+        pocet_13_pruniku = CI13[triangle_line].compute(IP13s);
+        ASSERT_DBG(pocet_13_pruniku < 3);
+//         DBGMSG("CI23: number of 1-3 intersections = %d\n",pocet_13_pruniku);
+        
+        for(unsigned int n=pocet_13_pruniku; n >= 1; n--){
+            IntersectionPointAux<1,3> IP (IP13s[IP13s.size()-n]);
+            
+            IntersectionPointAux<3,1> IP31 = IP.switch_objects();   // switch idx_A and idx_B and coords
+            IntersectionPointAux<3,2> IP32(IP31, triangle_line);    // interpolation uses local_bcoords_B and given idx_B
+            IntersectionPointAux<2,3> IP23 = IP32.switch_objects(); // switch idx_A and idx_B and coords back
+            
+            if( IP.dim_A() == 0 ) // if IP is vertex of triangle
+            {
+                // we are on line of the triangle, and IP.idx_A contains local node of the line
+                // E-E, we know vertex index
+                IP23.set_topology_A(RefElement<2>::interact(Interaction<0,1>(triangle_line))[IP.idx_A()], 0);
+            
+                if( IP.dim_B() < 3 )
+                    // if IP is on the surface of the tetrahedron,
+                    // we cannot trace it optimally
+                    intersection.pathologic_ = true;
```

Comment:
It seems that IntersectionAux<>::pathologic_ flag is never used neither through is_pathologic() method. If we do not use this flag we can also drop IntersectionPointAux::is_pathologic() and setting orientation to  2 in 1d-2d degenerate case.